source/loader/ze_loader.cpp

```diff
@@ -209,7 +209,8 @@ namespace loader
             return ZE_RESULT_ERROR_UNINITIALIZED;
         }
 
-        ze_result_t res = pfnInit(flags);
+        // Use the previously init ddi table pointer to zeInit to allow for intercept of the zeInit calls
+        ze_result_t res = globalInitStored->pfnInit(flags);
```

Comment:
is `globalInitStored` always guaanteed to be valid/non-null?

Reply:
Good point, globalInitStored is guranteed to be valid otherwise the GetProcTable api would have failed and we would not get to this code. But if we have a malformed driver the pfnInit could be nullptr, I fixed the previous check to ensure both the init from the driver and the table are not null.